### PR TITLE
Refactor to support symbols attribute and remove legacy functionality

### PR DESCRIPTION
**Pull request by stulees**
- All use of Firebase removed
- All logic refactored to now support `symbols` attribute and remove all unnecessary logic for retrieving/storing instruments
- Observing both `symbols` and `instrumentFields` attributes to trigger requesting new data
- Implemented sending _configured_ event to resolve race condition in template knowing when to send _start_ event, this now aligns with how rise-data-image does it also. 
- Demo updated to revise usage of component and remove import of Firebase scripts
- README updated to reflect accurate information from the refactoring

You can validate by scheduling http://widgets.risevision.com/staging/pages/2019.03.12.16.57/rise-data-financial/src/rise-data-financial.html as a URL item and test on ChrOS locally. The data should be logged in console.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,26 +133,23 @@ This component receives the following list of attributes:
 - **label**: ( string ): An optional label key for the text that will appear in the template editor. See 'Labels' section above.
 - **category**: "bonds" / "commodities" / "currencies" / "market statistics" / "stocks" / "world indexes". Required if the component is editable.
 - **symbols** ( string ): List of symbols separated by pipe symbol. Required if the component is editable. Example: "CADUSD=X|MXNUSD=X|USDEUR=X".
-- **financial-list** ( string / required ): Id of the financial list in Financial Selector ( won't be required once the component fully supports the symbols attribute ).
```
Comment (santiagonoguez) [inline README.md]: @stulees in the above symbols documentation, it should state that it is now required, and not required if

File: README.md [hunk 2]
```diff
@@ -146,26 +133,23 @@ This component receives the following list of attributes:
...
 - **instrument-fields** ( array of strings / optional ): if not provided, all fields will be retrieved. Example: ```["instrument", "name", "lastPrice"]```
 - **type**: 'realtime' ( default ) / 'historical'. See 'Historical Data' section above.
 - **duration**: Day / Week / 1M ( default ) / 3M / 6M / 1Y / 5Y. Only used when type == 'historical'. See 'Historical Data' section above.
-- **symbol** ( string ): Only used when type == 'historical'. indicates which instrument from the list of historical data needs to be retrieved. Example: 'AAPL.0'.
```
Comment (santiagonoguez) [inline README.md]: @stulees curious, this is left out because we now have an explicit symbol list ?

Comment (stulees) [reply README.md]: @santiagonoguez That's correct, a single `symbol` attribute serves no purpose when there is now a `symbols` attribute. If a designer only wants one symbol as the default, then they would simply apply for example `symbols=".ABC"` to get data for one symbol. Thanks

File: README.md (hunk 1 above)
Comment (stulees) [reply README.md]: @santiagonoguez I've updated to indicate `symbols` is required in the README, thanks

**Pull request merged**